### PR TITLE
docs: add a release guide

### PR DESCRIPTION
**Pull request by lucab**
This adds a step-by-step release guide.

### CODE REVIEW

**Comment by lucab** [comment]
@mike-nguyen can you please do a quick pass on this to make sure it is clear enough to understand and follow (and language is ok)?

File: docs/release-guide.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+* `unset RELEASE_VER`
+* `unset UPSTREAM_REMOTE`
+* `cargo clean`
+* `rm -rf vendor`
+* `git checkout master`
+* `git pull ${UPSTREAM_REMOTE} master`
+* `git push ${UPSTREAM_REMOTE} :release-${RELEASE_VER}`
+
+## Requirements
```
Comment (mike-nguyen) [inline docs/release-guide.md]: Can we move the `Requirements` section after the `Overview` section above?

File: docs/release-guide.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Release process
+
+This project uses [cargo-release][cargo-release] in order to prepare new releases, to tag and sign relevant git commit and to publish the resulting artifacts to [crates.io][crates-io].
```
Comment (mike-nguyen) [inline docs/release-guide.md]: This project uses [cargo-release][cargo-release] in order to prepare new releases, ~~to~~ tag and sign relevant git commit, and ~~to~~ publish the resulting artifacts to [crates.io][crates-io].

File: docs/release-guide.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+This document gives an high-level overview as well as a step-by-step guide on how to perform a release.
+
+## Overview
+
+Most of the process is automated with the help of cargo-release and its metadata entries in Cargo manifest.
+This helper is in charge of bumping the `version` field in the manifest in a dedicated commit, attaching the correspoding signed tag to it, and then producing another commit to prepare the project for the next development cycle.
+
+The two resulting commits can be then submitted in a dedicated PR and reviewed.
+Once merged, the last steps of the process consist in pushing the git tag, publishing the crate, and publishing the vendor tarball.
```
Comment (mike-nguyen) [inline docs/release-guide.md]: Once merged, the last steps of the process consist**s of** pushing the git tag, publishing the crate, and publishing the vendor tarball.

File: docs/release-guide.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+## Steps
+
+If this is the first time, make sure all the requirements from the section below are met.
```
Comment (mike-nguyen) [inline docs/release-guide.md]: If we move the `Requirements` section above the `Steps`, we won't need this statement.

**Review by mike-nguyen — COMMENTED** [review]
Looks good overall!  Some minor nits for en_IT :smile:  and moving the requirements section up.

File: docs/release-guide.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+In order to ease downstream packaging of Rust binaries, [cargo-vendor][cargo-vendor] is also used to provide an archive of vendored dependencies (only relevant for offline builds).
+
+The release process follows the usual PR-and-review flow, allowing an external reviewer to have a final check before publishing.
+
+This document gives an high-level overview as well as a step-by-step guide on how to perform a release.
+
+## Overview
+
+Most of the process is automated with the help of cargo-release and its metadata entries in Cargo manifest.
+This helper is in charge of bumping the `version` field in the manifest in a dedicated commit, attaching the correspoding signed tag to it, and then producing another commit to prepare the project for the next development cycle.
```
Comment (bgilbert) [inline docs/release-guide.md]: `corresponding`

File: docs/release-guide.md (hunk 3 above)
Comment (bgilbert) [reply docs/release-guide.md]: "steps" is plural, so it should be `consist of`.

**Comment by lucab** [comment]
Thanks both for proof-reading! I've amended with all your suggestions.

**Comment by lucab** [comment]
Unless further objections, I'm going to self-merge this tomorrow.

**Pull request merged**